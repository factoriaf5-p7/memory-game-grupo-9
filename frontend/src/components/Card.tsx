import React from "react";
// import pokeball from "./images/pokeball.png";
// import "./card.scss";

const Card = ({ onClick, card, index, isInactive, isFlipped, isDisabled }) => {

  // console.log(card)

  const handleClick = (e) => {
    !isFlipped && !isDisabled && onClick(index);
  };

  return (
    <div
      className={`card 
        ${isFlipped ? "is-flipped" : ''} 
        ${isInactive ? "is-inactive" : ''} 
        
      `}
      onClick={handleClick}
    >
      <div className="card-face card-font-face">
        <img src='/carta.jpg' alt="pokeball" />
        
      </div>
      <div className="card-face card-back-face">
      
        <img src={`img/${card.img}`} alt={`Imagen de ${card.name}`} />
      </div>
    </div>
  );
};

export default Card;