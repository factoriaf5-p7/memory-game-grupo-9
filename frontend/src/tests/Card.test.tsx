import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import Card  from '@/components/Card'

test('display card', async () => {
  // ARRANGE
  const testCard = { img:'batman.png', name: 'batman'}
  render(<Card card={testCard}   />)

  // ACT
  // await userEvent.click(screen.getByText('Load Greeting'))
  // await screen.findByRole('heading')

  // ASSERT
 expect(screen.getByAltText(/Imagen de/i)).toBeVisible(); 
  // expect(screen.getByRole('button')).toBeDisabled()
}); 

// test('card flips on click', () => {
//   // ARRANGE
//   render(<Card img="batman.png" name="hello there" toggle={vi.fn(() => true)}  />);

//   // ACT
//   const cardElement = screen.getByTestId(/card-/i);
//   userEvent.click(cardElement);


//   // ASSERT
//   const frontSide = screen.getByTestId('front-side');
//   // const bacreen.getBkText(/hola/i)Side = screen.getByTestId('back-side');
// // creen.getBText(/hola/i)
//   expect(screen.getByText(/hola/i)).toHaveTextContent('hola');
//   // expect(backSide).toContainElement(screen.getByAltText('Imagen de batman.png'));
// });

// test('card content displays correctly', () => {
//   // ARRANGE
//   render(<Card img="batman.png" name="hello there" />);

//   // ACT
//   const frontSide = screen.getByTestId('front-side');
//   const backSide = screen.getByTestId('back-side');

//   expect(frontSide).toHaveTextContent('Hola!');
//   expect(backSide).toContainElement(screen.getByAltText('Imagen de batman.png'));
// });