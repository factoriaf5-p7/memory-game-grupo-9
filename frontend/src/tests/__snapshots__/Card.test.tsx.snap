// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`card flips on click 1`] = `
<div
  class="card w-full h-full transition-transform duration-500 transform  bg-[url('carta.jpg')]"
>
  <div
    class="side flip front-face p-2 rounded-md shadow-md "
    data-testid="front-side"
  />
  hola
  <div
    class="side back-face bg-white p-2 rounded-md shadow-md hidden"
    data-testid="back-side"
  >
    <img
      alt="Imagen de hello there"
      class="w-full h-full"
      loading="lazy"
      src="/img/batman.png"
    />
  </div>
</div>
`;
